src/extension.ts

```diff
@@ -308,4 +315,46 @@ export async function activate(ctx: vscode.ExtensionContext) {
 
     explorer.refresh();
   }
+
+  async function pickExecutable() {
+    const picker = vscode.window.createQuickPick<TaskQuickPickItem>();
+    picker.busy = true;
+    picker.placeholder = "Select target to run.";
+    picker.show();
+
+    const runnableTasks = await getTasks('run');
+
+    picker.busy = false;
+    picker.items = runnableTasks.map(task => {
+      return {
+        label: task.definition.target,
+        detail: task.name,
+        description: "executable",
+        picked: false,
+        task: task,
+      }
+    });
+
+    return new Promise<TaskQuickPickItem>((resolve, reject) => {
+      picker.onDidAccept(() => {
+        const selection = picker.activeItems[0];
+        resolve(selection);
+        picker.dispose();
+      });
+      picker.onDidHide(() => reject());
+    });
+  }
+
+  async function runExecutable() {
+    let taskItem;
+    try {
+      taskItem = await pickExecutable();
+    } catch (err) {
+      // Pick cancelled.
+    }
+    if (taskItem != null) {
+      await vscode.tasks.executeTask(taskItem.task);
+    }
+    explorer.refresh();
```

Comment:
I don't think that's needed, I noticed when I worked on my previous patch that it's really costly to call that, I removed most of them already. Only needed in changeHandler AFAIK?